frontends/web/src/routes/settings/components/device-settings/firmware-setting.tsx

```diff
@@ -55,14 +54,21 @@ const FirmwareSetting = ({ deviceID, versionInfo, asButton = false }: TProps) =>
 
   return (
     <>
-      <SettingsItem
-        className={`${asButton ? styles.asButton : ''}`}
-        settingName={t('deviceSettings.firmware.title')}
-        secondaryText={secondaryText}
-        onClick={canUpgrade ? () => setDialogOpen(true) : undefined}
-        displayedValue={versionInfo.currentVersion}
-        extraComponent={extraComponent}
-      />
+      { asButton ? (
+        <Button
+          onClick={canUpgrade ? () => setDialogOpen(true) : undefined}
```

Comment:
nit: we can make a small fn out of this for btn onClick and use this here and also at the other button.

Reply:
done